### ISSUE TITLE
Common Modulus Attack Bug

### ISSUE THREAD

**Issue by jtsec92** [RsaCtfTool/RsaCtfTool#319]
When given the primitives `n`, `e1`, `e2`, `c1`, `c2` and passing them in via command line input for solving by running a command like:
```
./RsaCtfTool.py -n <n> -e <e1>,<e2> --uncipher <c1>,<c2>
```
The program will error out at `rsalibnum.py` with:
```
    m = int(introot(ct, g)[0])
TypeError: 'int' object is not subscriptable
```
However, changing the line from `m = int(introot(ct, g)[0])` to `m = int(introot(ct, g))` solves the problem. Is the `[0]` just there to handle large primitives (keys instead of n,e1,e2,c1,c2)?

**Comment by Ganapati** [comment]
Hi, thanks for this finding.
I have no time to patch it, can you make a pull request ?

**Comment by HEIDIHANSON** [comment]
> When given the primitives `n`, `e1`, `e2`, `c1`, `c2` and passing them in via command line input for solving by running a command like:
> ```
> ./RsaCtfTool.py -n <n> -e <e1>,<e2> --uncipher <c1>,<c2>
> ```
> The program will error out at `rsalibnum.py` with:
> ```
>     m = int(introot(ct, g)[0])
> TypeError: 'int' object is not subscriptable
> ```
> However, changing the line from `m = int(introot(ct, g)[0])` to `m = int(introot(ct, g))` solves the problem. Is the `[0]` just there to handle large primitives (keys instead of n,e1,e2,c1,c2)?
>$ssh-jeygen-t rsa-C.githubhanson.com

>2022-21-5 Heidi Hanson<

**Comment by saffarfe** [comment]
Yeah i've just wrote one from RsaCtfTool.py line 404 from 
    attackobj = RSAAttack(args)
To:
    attackobj = RSAAttack.RSAAttack(args)

**Comment by daedalus** [comment]
Solved in PR #332
